pallets/subtensor/src/staking.rs

```diff
@@ -95,6 +92,150 @@ impl<T: Config> Pallet<T> {
+        );
+        Self::deposit_event(Event::TakeDecreased(coldkey, hotkey, take));
+
+        // --- 6. Ok and return.
+        Ok(())
+    }
+
+    // ---- The implementation for the extrinsic increase_take
+    //
+    // # Args:
+    // 	* 'origin': (<T as frame_system::Config>::RuntimeOrigin):
+    // 		- The signature of the caller's coldkey.
+    //
+    // 	* 'hotkey' (T::AccountId):
+    // 		- The hotkey we are delegating (must be owned by the coldkey.)
+    //
+    // 	* 'take' (u16):
+    // 		- The stake proportion that this hotkey takes from delegations for subnet ID.
+    //
+    // # Event:
+    // 	* TakeDecreased;
+    // 		- On successfully setting a decreased take for this hotkey.
+    //
+    // # Raises:
+    // 	* 'NotRegistered':
+    // 		- The hotkey we are delegating is not registered on the network.
+    //
+    // 	* 'NonAssociatedColdKey':
+    // 		- The hotkey we are delegating is not owned by the calling coldket.
+    //
+    // 	* 'TxRateLimitExceeded':
+    // 		- Thrown if key has hit transaction rate limit
+    //
+    pub fn do_increase_take(
+        origin: T::RuntimeOrigin,
+        hotkey: T::AccountId,
+        take: u16,
+    ) -> dispatch::DispatchResult {
+        // --- 1. We check the coldkey signature.
+        let coldkey = ensure_signed(origin)?;
+        log::info!(
+            "do_increase_take( origin:{:?} hotkey:{:?}, take:{:?} )",
+            coldkey,
+            hotkey,
+            take
+        );
+
+        // --- 2. Ensure we are delegating a known key.
+        //        Ensure that the coldkey is the owner.
+        Self::do_take_checks(&coldkey, &hotkey)?;
+
+        // --- 3. Ensure we are strinctly increasing take
+        if let Ok(current_take) = Delegates::<T>::try_get(&hotkey) {
+            ensure!(take > current_take, Error::<T>::InvalidTake);
+        }
+
+        // --- 4. Ensure take is within the 0 ..= InitialDefaultTake (18%) range
+        let max_take = T::InitialDefaultTake::get();
+        ensure!(take <= max_take, Error::<T>::InvalidTake);
+
+        // --- 5. Enforce the rate limit (independently on do_add_stake rate limits)
+        let block: u64 = Self::get_current_block_as_u64();
+        ensure!(
+            !Self::exceeds_tx_delegate_take_rate_limit(
+                Self::get_last_tx_block_delegate_take(&coldkey),
+                block
+            ),
+            Error::<T>::TxRateLimitExceeded
+        );
+
+        // Set last block for rate limiting
+        Self::set_last_tx_block_delegate_take(&coldkey, block);
+
+        // --- 6. Set the new take value.
+        Delegates::<T>::insert(hotkey.clone(), take);
+
+        // --- 7. Emit the take value.
+        log::info!(
+            "TakeIncreased( coldkey:{:?}, hotkey:{:?}, take:{:?} )",
+            coldkey,
+            hotkey,
+            take
+        );
+        Self::deposit_event(Event::TakeIncreased(coldkey, hotkey, take));
+
+        // --- 8. Ok and return.
+        Ok(())
+    }
+
```

Comment:
If these both just set the `take` value, is there a reason they shouldn't be combined into a single extrinsic?